### ISSUE TITLE
Should remove the JqueryAsset registration from [yii\web\View::registerJs()]  ?

### ISSUE THREAD

**Issue by Maslow** [yiisoft/yii2#12588]
### What steps will reproduce the problem?

If I have to use JqueryAsset implemented by myself for a specified version , the yii\web\View::registerJs() method will register the built-in yii\web\JqueryAsset automatically !  Then I have two jquery.js in one Page ~~~~

ps: 
1.  I have to use Jquery for a specified version , because  some jquery plugins required.
2.  I always want to use yii\web\View::registerJs() as well.
3.  Now, I HAVE TO configure the assetsManager to solve my problem , it doesn't look smart.
### What is the expected result?

Just remove it ! 
Do not use JqueryAsset in yii\web\View::registerJs() to deal with the yii\web\View::POS_READY things.
Alternatively , just use NATIVE javascript instead. 
### What do you get instead?

 NATIVE javascript
### Additional info

Also consider to remove  [yii\web\Jquery] from yii2 ? 

| Q | A |
| --- | --- |
| Yii version | 2.0.9 |
| PHP version | PHP 7.0 |
| Operating system | OS X |

**Comment by SilverFire** [comment]
Use Dependecy Injection Container and override `yii\web\JqueryAsset` with your own.

`Yii::$container->set('yii\web\JqueryAsset', 'you\own\JqueryAsset`);`

See [Dependeny injection container](http://www.yiiframework.com/doc-2.0/guide-concept-di-container.html#practical-usage) guide page to know more.

**Comment by SilverFire** [comment]
Or even easier - override `bundles` array

http://www.yiiframework.com/doc-2.0/yii-web-assetmanager.html

**Comment by Maslow** [comment]
I know the two ways you said above,  but what I want to say is not that. 
For example , If I use View::registerJs() , but I don't want to use Jquery in my page.  Maybe I can't use it because of my BOSS ,  my Leader etc.  
This is what I want to express. 
Why hard-coding the Jquery in the View class?

**Comment by Maslow** [comment]
@SilverFire

**Comment by klimov-paul** [comment]
`JqueryAsset` is loaded **only** in case you are using `View::POS_READY` or `View::POS_LOAD` position for your JavaScript files. It is loaded because these positions require `JQuery` functions to be actually applied.

If you do not want JQuery o be involved in the process you should use other script positions.

**Comment by Maslow** [comment]
Thank you

**Comment by klimov-paul** [comment]
In general jQuery `$(document).ready()` statement can be replaced by pure JavaScript analog:

```js
document.addEventListener("DOMContentLoaded", function(event) { 
  //do work
});
```

**Comment by klimov-paul** [comment]
Resolved via #14865
